Apps/Admin/Client/Pages/PatientDetailsPage.razor.cs

```diff
@@ -0,0 +1,115 @@
+    /// Backing logic for the Patient Details page.
+    /// </summary>
+    public partial class PatientDetailsPage : FluxorComponent
+    {
+        [Inject]
+        private IDispatcher Dispatcher { get; set; } = default!;
+
+        [Inject]
+        private IState<MessageVerificationState> MessageVerificationState { get; set; } = default!;
+
+        [Inject]
+        private IState<PatientSupportState> PatientSupportState { get; set; } = default!;
+
+        [Inject]
+        private NavigationManager NavigationManager { get; set; } = default!;
+
+        private bool MessagingVerificationsLoading => this.MessageVerificationState.Value.IsLoading;
+
+        private bool HasMessagingVerificationsError => this.MessageVerificationState.Value.Error is { Message.Length: > 0 };
+
+        private IEnumerable<MessagingVerificationModel> MessagingVerifications => this.MessageVerificationState.Value.Data ?? Enumerable.Empty<MessagingVerificationModel>();
+
+        private bool PatientsLoaded => this.PatientSupportState.Value.Loaded;
+
+        private bool HasPatientsError => this.PatientSupportState.Value.Error is { Message.Length: > 0 };
+
+        private bool HasPatientsWarning => this.PatientSupportState.Value.WarningMessages.Any();
+
+        private PatientSupportDetails? Patient =>
+            this.PatientSupportState.Value.Result?.SingleOrDefault(x => x.Hdid == this.Hdid);
+
+        private string PatientName => StringManipulator.JoinWithoutBlanks(new[] { this.Patient?.PreferredName?.GivenName, this.Patient?.PreferredName?.Surname });
+
+        private string? Warning => this.Patient == null ? null : MapStatusToWarning(this.Patient.Status);
+
+        private string? Hdid { get; set; }
+
+        /// <inheritdoc/>
+        protected override void OnInitialized()
+        {
+            base.OnInitialized();
+
+            Uri uri = this.NavigationManager.ToAbsoluteUri(this.NavigationManager.Uri);
+            if (QueryHelpers.ParseQuery(uri.Query).TryGetValue("hdid", out StringValues hdid))
+            {
+                this.Hdid = hdid;
+                this.RetrievePatientData();
+            }
+            else
+            {
+                this.NavigationManager.NavigateTo("/support");
+            }
+        }
+
+        private static string? MapStatusToWarning(PatientStatus status)
+        {
+            return status switch
+            {
+                PatientStatus.NotFound => "Patient not found",
+                PatientStatus.Deceased => "Patient is deceased",
+                PatientStatus.NotUser => "Patient is not a user",
+                _ => null,
+            };
+        }
+
+        private void RetrievePatientData()
+        {
+            if (this.Patient?.Hdid != this.Hdid)
+            {
+                this.Dispatcher.Dispatch(new PatientSupportActions.ResetStateAction());
+                this.Dispatcher.Dispatch(new PatientSupportActions.LoadAction(PatientQueryType.Hdid, this.Hdid));
+            }
```

Comment:
This if statement can be simplified to `this.Patient == null`.

Reply:
Done